flink-tests/src/test/java/org/apache/flink/api/datastream/DataStreamBatchExecutionITCase.java

```diff
@@ -180,6 +191,107 @@ public void batchSumSingleResultPerKey() throws Exception {
         }
     }
 
+    /** Verifies that all broadcast input is processed before keyed input. */
+    @Test
+    public void batchKeyedBroadcastExecution() throws Exception {
```

Comment:
nit: How about we test with at least two keys?

Reply:
adding it to the test 👌